added PersonalShop page

tusharjkhunt opened the pull request:
I have used local image and hardcoded state values which will be replaced with database data.

tusharjkhunt committed:
added PersonalShop page

austinkwok1997 commented:
I think you might've forgotten to upload img1 and img7

bonnieli commented on client/src/pages/CreateProduct.js:
```diff
@@ -38,43 +48,43 @@ const useStyles = makeStyles((theme) => ({
-    value: 'Macarons',
-    label: 'Macarons',
+    value: "Macarons",
+    label: "Macarons",
```
looks like your linter config may be different from @austinkwok1997 you can create a prettier config file to stay consistent or sync to make sure your personal dev env is the same

tusharjkhunt commented:
> I think you might've forgotten to upload img1 and img7

Yes, images are missing. I thought of not pushing images to Github.

tusharjkhunt committed:
changes info has been added in PR description

tusharjkhunt committed:
changes info has been added in PR description

tusharjkhunt commented:
I have just pushed all the work inside this branch to avoid git-repo conflict.
Here are the details of the changes:
- prettier-config file added
- added 3 extra fields inside usermodel
- defined API route to Update a Shop details(name,disc,cover_pic)
- made few changes in register and login component(UI) to add protect router feature
- temporarily, all the protected routes have been redirected to "personal-shop" page instead of "home_dasboard" (only if authenticated, otherwise to login page)
- created necessary files (productContext, productState, productReducer) to handle product data
- integrated personal-shop page and all the products of this page are linked to product(galary)page, so we can view product directly from this page.
- important changes made in these files (productContext, productState, productReducer, PersonalShop, userModel, userRouter, Register, Login)  GN :)

tusharjkhunt committed:
added and integrated customized-shop-page

tusharjkhunt commented:
Here are the details of the new commit (6acf33c):
- added loading component
- fixed private route component issue
- handled load-user error by adding new case in authReducer.js file
- added links for "MY SHOP" & "MY ACCOUNT" button in the navbar
- created Customize Shop page (MyAccount.js)
- fetched & displayed data in Customize Shop page and integrated Customize Shop page with database and s3-bucket.

tusharjkhunt committed:
Merge branch 'dev' of https://github.com/hatchways/team-harvest-crunch into FE-PersonalShop-UI

tusharjkhunt committed:
loading user instantly everytime we make changes in database
